pyproject.toml

```diff
@@ -32,12 +32,37 @@ exclude = [
 ]
 
 [project.optional-dependencies]
-cudf = ["cudf>=23.08.00"]
+dev = [
+    "covdefaults>=2.3.0",
+    "duckdb>=1.1.1",
+    "hypothesis>=6.112.1",
+    "pandas>=2.0.3",
+    "polars>=1.8.1",
+    "pre-commit>=3.5.0",
+    "pyarrow>=17.0.0",
+    "pytest>=8.3.3",
+    "pytest-cov>=5.0.0",
+    "pytest-env>=1.1.5",
+    "pytest-randomly>=3.15.0",
+    "scikit-learn>=1.3.2",
+    "tqdm>=4.66.5",
+    "typing-extensions>=4.12.2",
+    "dask[dataframe]>=2023.5.0"
+]
```

Comment:
For running in ci/cd we should not pin any of these, as we want to have some old and random versions as well. Also dask requires python 3.9 or above.

Reply:
not an issue, was more of demonstration, will be fixed